src/config.ts

```diff
@@ -125,10 +125,11 @@ export interface SpecConfig {
    * Possible values:
    *  - 'immediate' is overriding top level elements only thus you can not append a new path or alter an existing value without erasing same level elements.
    *  - 'recursive' proceed to a deep merge and will concat every branches or override or create new values if needed. @see https://www.npmjs.com/package/merge
+   *  - 'deepmerge' uses deepmerge to merge which will concat object branches and concat arrays aswell @see https://www.npmjs.com/package/deepmerge
```

Comment:
uses `deepmerge` to merge, which will concat object branches and concat arrays as well